app/views/govuk_publishing_components/components/_step_by_step_nav_related.html.erb

```diff
@@ -35,12 +42,16 @@
                 data-track-label="<%= link[:href] %>"
                 data-track-dimension="<%= link[:text] %>"
                 data-track-dimension-index="29"
-                data-track-options='{"dimension96" : "<%= link[:tracking_id] %>" }'>
+                data-track-options='{"dimension96" : "<%= link[:tracking_id] %>" }'
+                <% if ga4_tracking %>
+                  data-ga4-link='{"event_name":"navigation", "type":"related content", "index": "{"index_link":"<%= index %>"}", "index_total":"<%= links.length %>"}'
```

Comment:
Same question here. An alternative might be to define the data object prior to the link, e.g.

```Ruby
<% links.each_with_index do |link, index| %>
  <%
    data = {
      track-category: "stepNavPartOfClicked",
      track-action: pretitle,
      # (etc.)
    }
    data[:ga4_link] = { etc. } if ga4_tracking
  %>
  <%= link_to link[:href], data, classes %>
<% end %>
```

(this pseudo code is full of errors but hopefully clear)

Reply:
Thanks Andy. I’ve just removed the quotes for now but let me know if you’d prefer to have the above implemented 👍

Reply:
No that's fine. As long as it works in a real use case.